shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -0,0 +1,186 @@
+package io.flutter.embedding.engine.android;
+
+import android.support.annotation.NonNull;
+import android.view.MotionEvent;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+
+/**
+ * Sends touch information from Android to Flutter in a format that Flutter
+ * understands.
+ */
+public class AndroidTouchProcessor {
+
+  // Must match the PointerChange enum in pointer.dart.
+  // TODO(mattcarroll): convert these to an IntDef
+  private static final int POINTER_CHANGE_CANCEL = 0;
+  private static final int POINTER_CHANGE_ADD = 1;
+  private static final int POINTER_CHANGE_REMOVE = 2;
+  private static final int POINTER_CHANGE_HOVER = 3;
+  private static final int POINTER_CHANGE_DOWN = 4;
+  private static final int POINTER_CHANGE_MOVE = 5;
+  private static final int POINTER_CHANGE_UP = 6;
+
+  // Must match the PointerDeviceKind enum in pointer.dart.
+  // TODO(mattcarroll): convert these to an IntDef
+  private static final int POINTER_DEVICE_KIND_TOUCH = 0;
+  private static final int POINTER_DEVICE_KIND_MOUSE = 1;
+  private static final int POINTER_DEVICE_KIND_STYLUS = 2;
+  private static final int POINTER_DEVICE_KIND_INVERTED_STYLUS = 3;
+  private static final int POINTER_DEVICE_KIND_UNKNOWN = 4;
+
+  // Must match the unpacking code in hooks.dart.
+  private static final int POINTER_DATA_FIELD_COUNT = 19;
+  private static final int BYTE_PER_FIELD = 8;
+
+  @NonNull
+  private final FlutterRenderer renderer;
+
+  /**
+   * Constructs an {@code AndroidTouchProcessor} that will send touch event data
+   * to the Flutter execution context represented by the given {@link FlutterRenderer}.
+   */
+  // TODO(mattcarroll): consider moving packet behavior to a FlutterInteractionSurface instead of FlutterRenderer
+  public AndroidTouchProcessor(@NonNull FlutterRenderer renderer) {
+    this.renderer = renderer;
+  }
+
+  /**
+   * Sends the given {@link MotionEvent} data to Flutter in a format that
+   * Flutter understands.
+   */
+  public boolean onTouchEvent(MotionEvent event) {
```

Comment:
If there's documentation for how we're organizing these packets, it should be linked here.

If there isn't documentation for how we're organizing these packets, you should write it while you still remember how this works.